### ISSUE TITLE
[Bug?] Paint Replacing Other Hotbar Items

### ISSUE THREAD

**Issue by piotmetans** [tsunamayo/Starship-EVO#5421]
This is an old one, but waaaay back paint used to behave a bit differently when selected through the middle-mouse quick menu. If your hot-bars already had paint in them somewhere, selecting a paint with the quick menu from something in the world would automatically switch to that paint option that was already in your hot-bars, as well as swap to the color you selected. I believe this ended up changing inadvertently when the multi-stacks were implemented. Currently any quick-menu paint selection simply replaces whatever the current hot-bar item is.

Not sure if this was an intended change or not, but I thought I'd make this in case it isn't. I always liked it because it requires less hot-bar space and I tend to pick shapes almost exclusively from the quick menu anyway.

24w02a

**Comment by ProPeach** [comment]
Oh yeah, this one changed a few times lol. I remember making a report to ensure the behaviour where existing paint on your hotbar is found instead of replacing your current item as exactly this happened before

**Comment by tsunamayo** [comment]
@piotmetans Hi, I do not reproduce what you are explaining. For me picking a paint will select an existing one if present, otherwise it will create a new paint. @ProPeach How is it working on your side?
Cheers

**Comment by piotmetans** [comment]
Sorry for the late reply! I think I'm doing a bad job explaining, maybe this will help.

https://github.com/tsunamayo/Starship-EVO/assets/660015/4eb9f8ec-170c-4496-8598-1da44e2a4372

The first 3 selections here are demonstrating current behavior with a paint selected on the hotbar. You can see that it will always replace the paint with whatever paint is selected from the quick-menu.

I then swapped to a shield block and quick-selected the paint that was already on my bars. It automatically swaps to the paint option for me instead of replacing the block.

However, when quick-selecting the paint that was not on my bars, it changes the shield block into the paint instead.

Note that my other hotbars were completely filled up for this demonstration.

~~

Basically, in the yellow paint scenario above, the yellow paint would replace the black paint instead of the shield block in previous versions of the game. New paints currently always take priority and will replace any and every item in your bars.

I just wanted to clarify whether this was intended behavior or if it was an unintended bug. If it is indented, I'd like to propose re-implementing the older behavior because it helps save on hotbar space _immensly_ and keeps them relatively tidy. I can't tell you how many times I've messed up my hotbar accidentally quick-selecting a paint before swapping to a paint on my bars.

Hopefully this is easier to follow!

**Comment by tsunamayo** [comment]
@piotmetans Thanks it is more clear now. You wanted me to re-introduce a bug because that was more convenient ^^
So I have added a new rule, if the item type of the item picked is different than the current item selected it will try to select an item of the same type to replace from your hotbar.
For 24w05a.
Cheers
